addons/mail/models/mail_message.py

```diff
@@ -884,11 +884,13 @@ def _message_format(self, fnames, format_reply=True):
                 'id': message_sudo.author_guest_id.id,
                 'name': message_sudo.author_guest_id.name,
             } if message_sudo.author_guest_id else [('clear',)]
+            thread_record_sudo = False
             if message_sudo.model and message_sudo.res_id:
                 record_sudo = self.env[message_sudo.model].browse(message_sudo.res_id).sudo()
                 record_name = record_sudo.with_prefetch(thread_ids_by_model_name[message_sudo.model]).display_name
                 # if not a thread, display_name is the default subject
-                default_subject = record_sudo._message_compute_subject() if hasattr(record_sudo, '_message_compute_subject') else record_name
+                thread_record_sudo = record_sudo if self.env['ir.model']._get(message_sudo.model).is_mail_thread else False
```

Comment:
Isn't that change adding queries ?

Reply:
I don't really like relying on hasattr but yes technically it does add a query :/

Will revert and add another hasattr later to be technically complete.